encoding/codecv1.go

```diff
@@ -456,3 +457,38 @@ func (d *DACodecV1) computeBatchDataHash(chunks []*Chunk, totalL1MessagePoppedBe
 	dataHash := crypto.Keccak256Hash(dataBytes)
 	return dataHash, nil
 }
+
+// BlobDataProofFromBlobBytes calculates a blob's challenge digest, commitment, and proof from blob bytes.
+func (d *DACodecV1) BlobDataProofFromBlobBytes(blobBytes []byte) (common.Hash, kzg4844.Commitment, kzg4844.Proof, error) {
+	blob, err := makeBlobCanonical(blobBytes)
+	if err != nil {
+		return common.Hash{}, kzg4844.Commitment{}, kzg4844.Proof{}, fmt.Errorf("failed to convert blobBytes to canonical form: %w", err)
+	}
+
+	commitment, err := kzg4844.BlobToCommitment(blob)
+	if err != nil {
+		return common.Hash{}, kzg4844.Commitment{}, kzg4844.Proof{}, fmt.Errorf("failed to create blob commitment: %w", err)
+	}
+	blobVersionedHash := kzg4844.CalcBlobHashV1(sha256.New(), &commitment)
+
+	challengeDigest := crypto.Keccak256Hash(crypto.Keccak256(blobBytes), blobVersionedHash[:])
```

Comment:
Is this correct? I thought the `challengeDigest` would be computed differently before CodecV7

Reply:
yes. It's correct. The differences are only introduced in v7.

Reply:
wait why is da-codec@v1 being updated?